src/components/OfflineIndicator.tsx

```diff
@@ -0,0 +1,56 @@
+import React from 'react';
+import {StyleProp, View, ViewStyle} from 'react-native';
+import {OnyxEntry} from 'react-native-onyx/lib/types';
+import styles from '@styles/styles';
+import variables from '@styles/variables';
+import type {Network} from '@src/types/onyx';
+import Icon from './Icon';
+import * as Expensicons from './Icon/Expensicons';
+import type {LocaleContextProps} from './LocaleContextProvider';
+import {withNetwork} from './OnyxProvider';
+import Text from './Text';
+import withLocalize from './withLocalize';
+import withWindowDimensions from './withWindowDimensions';
+import type {WindowDimensionsProps} from './withWindowDimensions/types';
+
+type OfflineIndicatorProps = LocaleContextProps &
+    WindowDimensionsProps & {
+        /** Information about the network */
+        network: OnyxEntry<Network>;
+
+        /** Optional styles for container element that will override the default styling for the offline indicator */
+        containerStyles?: StyleProp<ViewStyle>;
+
+        /** Optional styles for the container */
+        style?: StyleProp<ViewStyle>;
+    };
+
+const setStyles = (containerStyles: StyleProp<ViewStyle>, isSmallScreenWidth: boolean): StyleProp<ViewStyle> => {
+    if (!!containerStyles && ((Array.isArray(containerStyles) && containerStyles.length) || Object.keys(containerStyles).length)) {
+        return containerStyles;
+    }
```

Comment:
Maybe it makes sense to use lodash isEmpty here?
```suggestion
  if (!isEmpty(containerStyles)) {
        return containerStyles;
    }
```

Reply:
Yeah in my opinion here is a good case to use lodash isEmpty , or we can create our own implementation for this if we would like to get rid of lodash in future

Reply:
Yes, that would be a good case for `isEmpty`, but I was analyzing this piece of code and I think the JS implementation was only to assert if `containerStyles` was defined or not, so I think we could simplify here:

```ts
const setStyles = (containerStyles: StyleProp<ViewStyle> | undefined, isSmallScreenWidth: boolean): StyleProp<ViewStyle> => {
    if (containerStyles) {
        return containerStyles;
    }
    return isSmallScreenWidth ? styles.offlineIndicatorMobile : styles.offlineIndicator;
};
```